source/common/stats/symbol_table_impl.h

```diff
@@ -173,6 +176,9 @@ class SymbolTableImpl : public SymbolTable {
   // This must be held during both encode() and free().
   mutable Thread::MutexBasicLockable lock_;
 
+  // This must be held while updating stat_name_sets_.
```

Comment:
For simplicity thoughts on just using the lock you already have defined? Do we really need 2 locks?